src/com/dre/brewery/listeners/WorldListener.java

```diff
@@ -45,7 +45,7 @@ private void lwDataTask(World world) {
 
 	@EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
 	public void onWorldUnload(WorldUnloadEvent event) {
-		DataSave.save(true);
+		DataSave.save(P.p.shuttingDown);
 		String worldName = event.getWorld().getName();
 		Barrel.onUnload(worldName);
```

Comment:
Have you tested this? In line 50 it removes all Barrel Data of the unloading world, meaning in the async task, that runs later, the Barrel data is already deleted and can't be saved.

Reply:
Ah, in my case the unloading world(s) don't have barrels/data. I suppose we could wrap most of this async?